core/extension/gdextension_interface.h

```diff
@@ -1383,6 +1384,20 @@ typedef GDExtensionVariantFromTypeConstructorFunc (*GDExtensionInterfaceGetVaria
  */
 typedef GDExtensionTypeFromVariantConstructorFunc (*GDExtensionInterfaceGetVariantToTypeConstructor)(GDExtensionVariantType p_type);
 
+/**
+ * @name variant_get_ptr_internal
+ * @since 4.4
+ *
+ * Returns a function to retrieve a pointer to a variant's internal value.
+ * Each function assumes the variant's type has already been determined and matches the function.
+ * If any function is called with a variant of mismatching type, the behavior is undefined and may result in a segfault.
```

Comment:
I think this is a great paragraph, but as mentioned in my comments, I would maybe add something along:

> This function is particularly useful in situations where you know the variant type upfront, and can thus directly fetch the pointer of the correct type. It can make sense to cache the returned function pointers.

I don't have a great wording, but the idea is to convey the idea behind this function existing despite other similar APIs. People won't have access to the information discussed in this PR anymore.